packages/babel-plugin-transform-react-constant-elements/src/index.js

```diff
@@ -18,6 +18,19 @@ export default declare((api, options) => {
   // Element -> Target scope
   const HOISTED = new WeakMap();
 
+  function declares(
+    node: t.ThisExpression | t.Identifier | t.JSXIdentifier,
+    scope,
+  ) {
+    // Ideally this should always be a ThisExpression 🤷
+    if (t.isThisExpression(node) || t.isJSXIdentifier(node, { name: "this" })) {
```

Comment:
I think we'll also need this for `super`, `new.target`, `arguments`, etc? I think it'd be better to handle an `JSXOpeningElement` that check if we're referencing a binding or using a lowercase name.

Reply:
Ahh, I see we added `JSXIdentifier` to `ReferencedIdentifier`, so this should be ok.